src/lib/Stage.class.ts

```diff
@@ -55,30 +61,44 @@ class Stage extends EventEmitter {
   }
 
   public run(): void {
-    let quadCount = 0
-    // let iteratorCount = 0
-    const writer = new Writer(this.destination(), { end: false, format: 'N-Triples' })
-
-    this.generator.on('data', quad => {
-      writer.addQuad(quad)
-      quadCount ++
-      this.emit('generatorResult', quadCount)
-    })
-
-    this.generator.on('error', e => {
-      this.emit('error', e)
-    })
-
-    this.iterator.on('data', $this => {
-      this.generator.run($this)
-      this.emit('iteratorResult', $this)
-    })
-
-    this.iterator.on('error', e => {
-      this.emit('error', e)
-    })
-    this.iterator.run()
-  }
+    const writer = new Writer(this.destination(), { end: false, format: 'N-Triples' });
+    let quadCount = 0;
+
+    this.generators.forEach(generator => {
+      generator.on('data', (quad) => {
+          writer.addQuad(quad);
+          quadCount++;
+          this.emit('generatorResult', quadCount);
+      });
+
+      generator.on('end', (iterationsIncoming, statements, processed) => {
+        this.totalProcessed += processed
+        if (this.totalProcessed >= (iterationsIncoming * this.configuration.generator.length)){
```

Comment:
Here you reference `this.configuration.generator`, while this is part of a `forEach` loop over `this.generators`. Is that intentional?

Reply:
yes, since I want to know the length (the number of generators) for this stage, since another stage might have a different number of generators, hence I look for the stages configuration to see how many generators are present.